pkg/kubelet/pod_workers.go

```diff
@@ -765,12 +797,18 @@ func (p *podWorkers) managePodLoop(podUpdates <-chan podWork) {
 			klog.ErrorS(err, "Error syncing pod, skipping", "pod", klog.KObj(pod), "podUID", pod.UID)
 
 		case update.WorkType == TerminatedPodWork:
+			klog.V(4).InfoS("Processing pod event done", "pod", klog.KObj(pod), "podUID", pod.UID, "updateType", update.WorkType)
+			if p.queueReservedWork(pod) {
```

Comment:
The usage of a method that returns a bool that is not named 'has*|should*' is a bit unusual.  I'm confused though - we need to wake up *another* worker (because the queued work isn't guaranteed to have our uid, in fact is very likely not to), so I expected this chunk to simply queue an action on another pod's work queue.  That implies that we start one work queue for every incoming uid, and for a given uid we may have updates for that uid be "no-ops" until the previous duplicate is freed.

Also, many pods can attempt to use the same full name, so I expected the reserved work for a pod name to be a list, not a single entry.  

Finally, if a pod is blocked long enough on shutdown I think a reason is going to have to be provided to pending pod status.  I'm a little uncomfortable about having that in this method, so it might simply be a new sync method (`syncWaitingPod`) or something provided to this method.

Reply:
> Finally, if a pod is blocked long enough on shutdown I think a reason is going to have to be provided to pending pod status. I'm a little uncomfortable about having that in this method, so it might simply be a new sync method (`syncWaitingPod`) or something provided to this method.

@smarterclayton 
Can you please elaborate on this? 
I am not sure where to call the method that sets a reason for the pending pod status.
Do we want to give such work to the pending pod's goroutine?